site/assets/scss/_search.scss

```diff
@@ -33,30 +17,68 @@
     margin-left: -140px;
   }
 
-  .form-control {
-    padding-right: 2.75rem;
-    color: $white;
-    background-color: rgba($black, .1);
-    border-color: rgba($white, .4);
-    transition-property: background-color, border-color, box-shadow;
+}
 
-    &::placeholder {
-      color: rgba($white, .65);
-    }
+.DocSearch-Container {
+  z-index: 1030;
 
-    &::-webkit-search-cancel-button {
-      appearance: none;
-      width: 1rem;
-      height: 1rem;
-      cursor: pointer;
-      background: escape-svg($search-clear-icon) no-repeat 0 0;
-      background-size: 100% 100%;
-    }
+  @include media-breakpoint-up(lg) {
+    padding-top: 4rem;
+  }
+}
+
+.DocSearch-Button {
+  --docsearch-searchbox-background: #{rgba($black, .1)};
+  --docsearch-searchbox-color: #{$white};
+  --docsearch-searchbox-focus-background: #{rgba($black, .25)};
+  --docsearch-searchbox-shadow: #{0 0 0 .25rem rgba($bd-accent, .4)};
+  --docsearch-text-color: #{$white};
+  --docsearch-muted-color: #{rgba($white, .65)};
 
-    &:focus {
-      background-color: rgba($black, .25);
-      border-color: rgba($bd-accent, 1);
-      box-shadow: 0 0 0 .25rem rgba($bd-accent, .4);
+  width: 100%;
+  margin: 0;
+  border: 1px solid rgba($white, .4);
+  @include border-radius(.375rem);
+
+  .DocSearch-Search-Icon {
+    opacity: .65;
+  }
+
+  &:active,
+  &:focus,
+  &:hover {
+    border-color: rgba($bd-accent, 1);
+
+    .DocSearch-Search-Icon {
+      opacity: 1;
     }
   }
 }
+
+.DocSearch-Button-Keys {
+  min-width: 0;
+  padding-right: .25rem;
+  padding-left: .25rem;
+  background: rgba($black, .25);
+  @include border-radius(.25rem);
+}
+
+.DocSearch-Button-Key {
+  top: 0;
+  width: auto;
+  height: 1.25rem;
+  padding-right: .125rem;
+  padding-left: .125rem;
+  margin-right: 0;
+  font-size: .875rem;
+  background: none;
+  box-shadow: none;
+}
+
+.DocSearch-Commands-Key {
+  padding-left: 1px;
+  font-size: .875rem;
+  background-color: rgba($black, .1);
+  background-image: none;
+  box-shadow: none;
+}
```

Comment:
Maybe not a priority if you are more focused on the new DocSearch integration itself but the vertical alignment of the icons on the left is a bit off:

![Screenshot from 2022-04-21 18-00-32](https://user-images.githubusercontent.com/17381666/164503031-b6f87d82-1ec2-4077-98fd-39f990e0ff0b.png)
![Screenshot from 2022-04-21 18-00-22](https://user-images.githubusercontent.com/17381666/164503033-25e91f58-dc49-466c-b1b3-6cee902e487a.png)

The following rule seems to offer a better rendering:

```scss
.DocSearch-Hit-Container {
  .DocSearch-Hit-icon {
    svg {
      vertical-align: top;
    }
  }
}
```

Reply:
On Chrome, the highlighted letters in violet are not well vertically aligned:

![Screenshot from 2022-04-21 18-28-05](https://user-images.githubusercontent.com/17381666/164507417-dc3fc24e-1878-4d95-9395-1f2fb46f22af.png)

A fix with no space at all could be:

```scss
.DocSearch-Modal {
  mark {
    padding: 0;
  }
}
```

Otherwise with `padding-top: 0; padding-bottom: 0`.